\documentclass[final]{beamer}
\usepackage{grffile}
\mode<presentation>{\usetheme{CambridgeUSPOL}}

\usepackage[utf8]{inputenc}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{graphicx}
\usepackage{array,booktabs,tabularx}
\usepackage{colortbl, xcolor}
\newcolumntype{Z}{>{\centering\arraybackslash}X}

% rysunki
\usepackage{tikz}
\usepackage{ifthen}
\usepackage{xxcolor}
\usetikzlibrary{arrows}
\usetikzlibrary[topaths]
\usetikzlibrary{decorations.pathreplacing}
%\usepackage{times}\usefonttheme{professionalfonts}  % times is obsolete
\usefonttheme[onlymath]{serif}
\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                       % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

\usecolortheme{seagull}
\useinnertheme{rectangles}
\setbeamercolor{item projected}{bg=darkred}
% \setbeamertemplate{enumerate items}[default]
\setbeamertemplate{caption}{\insertcaption} 
\setbeamertemplate{navigation symbols}{}
\setbeamercovered{transparent}
\setbeamercolor{block title}{fg=darkred}
\setbeamercolor{local structure}{fg=darkred}

\setbeamercolor*{enumerate item}{fg=darkred}
\setbeamercolor*{enumerate subitem}{fg=darkred}
\setbeamercolor*{enumerate subsubitem}{fg=darkred}

\setbeamercolor*{itemize item}{fg=darkred}
\setbeamercolor*{itemize subitem}{fg=darkred}
\setbeamercolor*{itemize subsubitem}{fg=darkred}

\newlength{\columnheight}
\setlength{\columnheight}{96cm}
\renewcommand{\thetable}{}
\def\andname{,}
\authornote{}

\renewcommand{\APACrefatitle}[2]{}
\renewcommand{\bibliographytypesize}{\footnotesize} 
\renewcommand{\APACrefYearMonthDay}[3]{%
  {\BBOP}{#1}
  {\BBCP}
}

\begin{document}
\SweaveOpts{concordance=TRUE}



<<knitrIntro, echo = FALSE, message=FALSE>>=
library(dplyr)
library(ggplot2)
library(gridExtra)
library(xtable)

size_mod <- 4
cool_theme <- theme(plot.background=element_rect(fill = "transparent",
                                                 colour = "transparent"),
                    panel.grid.major = element_line(colour="lightgrey", linetype = "dashed"),
                    panel.background = element_rect(fill = "white",colour = "black"),
                    legend.background = element_rect(fill="NA"),
                    legend.position = "bottom",
                    axis.text = element_text(size=12 + size_mod),
                    axis.title.x = element_text(size=16 + size_mod, vjust = -1), 
                    axis.title.y = element_text(size=16 + size_mod, vjust = 1),
                    strip.text = element_text(size=17 + size_mod, face = "bold"),
                    legend.text = element_text(size=13 + size_mod), 
                    legend.title = element_text(size=17 + size_mod),
                    plot.title = element_text(size=20 + size_mod))
load("poster_data.RData")
load("report4.RData")
@


\date{}
\author{Micha\l{} Burdukiewicz\inst{1}, Piotr Sobczyk\inst{2}, Pawe\l{} Mackiewicz\inst{1} and Ma\l{}gorzata Kotulska\inst{3}
}


\institute{\small{\textsuperscript{1}University of Wroc\l{}aw, Department of Genomics, Poland 

\vspace{0.3cm}

\textsuperscript{2}Wroc\l{}aw University of Technology, Department of Mathematics, Poland

\vspace{0.3cm}

\textsuperscript{3}Wroc\l{}aw University of Technology, Department of Biomedical Engineering, Poland}
}
}
\title{\huge N-gram analysis of amyloid data}

\begin{frame}
\begin{columns}
\begin{column}{.49\textwidth}
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}
\parbox[t][\columnheight]{\textwidth}
{
\begin{block}{Aim}
Investigate putative motifs of hot-spots, short sequences responsible for amyloidogenicity which is associated with a number of clinical disorders, for example Alzheimer's or Creutzfeldt-Jakob's diseases.
\end{block}
\vfill


\begin{block}{AmyLoad database}
<<AmyLoad, echo = FALSE, message=FALSE, fig.width=14>>=
totals <- seq_inter_ct %>% group_by(tar) %>% summarize(total = sum(Freq)) %>% select(total) %>% unlist
@

The sequences used in the study (\Sexpr{totals[1]} non-amyloids and \Sexpr{totals[2]} amyloids) were extracted from AmyLoad database~\citep{wozniak_amyload:_2015}.

<<AmyLoadPlot, echo = FALSE, message=FALSE, fig.width=14>>=
ggplot(seq_inter_ct, aes(x = len, y = prop, fill = tar, label = paste0(round(prop, 4) * 100, "%"))) +
  geom_bar(position = "dodge", stat = "identity") +
  scale_fill_manual("Amyloid", values = c("tan2", "skyblue")) +
  scale_x_discrete("Length") +
  scale_y_continuous("Count") + 
  geom_text(position = position_dodge(width = 0.9), vjust=-0.25) +
  cool_theme #+ theme(legend.position = "right")
@


\end{block}
\vfill

\begin{block}{Clustering of amino acids}

\begin{enumerate}[1.]
\item Nine scales representing properties important in the amylogenicity: hydrophobicity, size polarity and solvent accessibility from AAIndex database~\citep{kawashima_aaindex:_2008} were chosen. Additionally, two frequencies of forming contact sites~\citep{wozniak_characteristics_2014} were added. Scales were normalized between 0 i 1.
\item All combinations of characteristics (only one scale per property) were clustered using Euclidean distance and Ward's method.
\item Each clustering was divided into 3 to 6 groups creating 144 encodings of amino acids.
\item Redundant 51 encodings (identical to other encodings) were removed.
\end{enumerate}

\end{block}
\vfill

\begin{block}{Evaluation}

\begin{enumerate}[1.]
\item Sequences shorter than 6 amino acids were discarded.
\item From each sequence 5-grams were extracted. All n-grams were labelled as their sequence of the origin (e.g. 5-grams extracted from amyloid sequence were labelled as positive).
\item For each encoding features were filtered by the QuiPT and used to train the Random Forests~\citep{liaw_classification_2002}. The procedure was performed on three traning sets: a) 6 amino acids, b) 10 amino acids or shorter, c) 15 amino acids or shorter creating three classifiers.
\item All classifiers were evaluated in the 5-fold cross-validation. The sequence was labelled as positive if at least one 5-gram was assessed as amylogenic.
\item The cross-validation was repeated 8 times.
\end{enumerate}

\end{block}
\vfill

\begin{block}{Encoding distance}

The encoding distance between \textbf{A} and \textbf{B} is defined as the minimum number of amino acids that have to be shifted between subgroups of encoding \textbf{A} to make it identical to \textbf{B} (order of subgroups in the encoding and amino acids in a group is unimportant).

\begin{columns}
\column{.4\textwidth}
<<encodingDistanceA, echo = FALSE, message=FALSE, results='asis'>>=
enc2df <- function(x)
  data.frame(Group = names(x), Elements = sapply(x, paste0, collapse = ", "))
l1 <- list('1' = c("a", "b", "c"),
             '2' = c("d", "e"))
l2 <- list('1' = c("a", "b"),
             '2' = c("d", "e"),
            '3' = c("c"))

print(xtable(enc2df(l1), "Encoding \\textbf{A}."), include.rownames = FALSE)
@
\column{.4\textwidth}

<<encodingDistanceB, echo = FALSE, message=FALSE, results='asis'>>=
print(xtable(enc2df(l2), "Encoding \\textbf{B}."), include.rownames = FALSE)
@
\end{columns}

The encoding distance between \textbf{A} and \textbf{B} is 1 (element \textit{c} must be moved from Group 3 to Group 1).

\end{block}
\vfill



}
\end{minipage}
\end{beamercolorbox}
\end{column}


%new column ------------------------------------------------------    

\begin{column}{.49\textwidth}
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}  
\parbox[t][\columnheight]{\textwidth}
{




\begin{block}{Specificity versus sensitivity}
<<specSensTab, echo = FALSE, message=FALSE, results='asis'>>=
tmp <- res %>% group_by(len) %>% filter(mSpec == max(mSpec) | mSens == max(mSens))
colnames(tmp) <- c("Training length", "Encoding ID", "AUC", "Specificity", "Sensitivity", "Number of groups")

rws <- seq(1, nrow(tmp) - 1, by = 2)
col <- rep("\\rowcolor[gray]{0.75}", length(rws))

print(xtable(tmp, "Encodings with the best sensitivity and specificity for each training set type.", digits = 4), include.rownames = FALSE, booktabs = TRUE, add.to.row = list(pos = as.list(rws), command = col))
@

<<specSensPlot, echo = FALSE, message=FALSE, fig.width=14, fig.height=18>>=
set.seed(2)
ggplot(ss_dat, aes(x = speced, y = sensed, size = mSpec, colour = mSens, shape = bestAUC)) +
  geom_tile(colour = "black", fill = "lightgrey", size = 1) +
  geom_point(position = position_jitter(0.27, 0.27)) +
  facet_wrap(~ len, ncol = 1) +
  scale_colour_gradientn("Mean sensitivity", colours = c("blue", "yellow", "red")) +
  cool_theme +
  scale_size_continuous("Mean specificity", range = c(1, 5)) +
  scale_x_continuous("Encoding distance (best specificity)") + 
  scale_y_continuous("Encoding distance (best sensitivity)") +
  scale_shape_manual("AUC", values = c(16, 1))
@

\end{block}
\vfill

\begin{block}{Bibliography}
  \tiny{
  \bibliographystyle{apalike}
  \bibliography{amyloids}
  }
  \end{block}
  \vfill

}
\end{minipage}
\end{beamercolorbox}
\end{column}
\end{columns}  
\end{frame}
\end{document}